corehq/apps/accounting/static/accounting/js/payment_method_handler.js

```diff
@@ -137,7 +137,7 @@ hqDefine('accounting/js/payment_method_handler', [
                 const containerId = 'stripe-card-container';
                 if (show) {
                     if (document.getElementById(containerId)) {
-                        cardElement.mount('#' + containerId);
+                        cardElement.mount('#' + formId + ' #' + containerId);
```

Comment:
there are multiple objects with the same container id on the page? it's invalid HTML to have that scenario. it feels more like a bandaid than an actual fix. I would like to hear if @nospame has additional context about these other modals. if not perhaps we should either remove them or fix the multiple similar ids on the same page issue?

Reply:
The `containerId` is the stripe container, I can update it to use a class instead. I don't think it's necessarily weird to have multiple modals on the page.

Reply:
it's not multiple modals i'm concerned about. it's that they all have the same element id... that's the issue

Reply:
I say this because if we have to specify `#some-form-id #some-modal-id` because `#some-modal-id` is not specific enough...then `#some-modal-id` should be more specific. `#foo` should be the final level of specificity for selecting any element by id.

Reply:
Updated this logic in 1b027b949d3d61813972989dd50cd9fb9090c619